### PR TITLE
[INLONG-10841][SDK] Transform SQL supports toBase64 function

### LINKED ISSUE #10841: [Feature][SDK] Transform SQL supports toBase64 function
### Description

- parent https://github.com/apache/inlong/issues/10796

to_base64(str): Return the base64-encoded result from string.
  

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zoy0**
Fixes #10841

### Motivation

Add a string function class：ToBase64Function.
Add the corresponding unit test codes

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * parse
+     * @param sourceData
+     * @param rowIndex
+     * @return
+     */
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObj = stringParser.parse(sourceData, rowIndex, context);
+        String string = OperatorTools.parseString(stringObj);
+        return Base64.getEncoder().encodeToString(string.getBytes(StandardCharsets.UTF_8));
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java:56]: Not need to create an encoder every time.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * description: to_base64(string1)--returns the base64-encoded result from string1
+ */
+public class ToBase64Function implements ValueParser {
+
+    private final ValueParser stringParser;
+
+    /**
+     * Constructor
+     * @param expr
+     */
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java:41]: add more information or remove it

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public ToBase64Function(Function expr) {
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    /**
+     * parse
+     * @param sourceData
+     * @param rowIndex
+     * @return
+     */
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java:51]: ditto

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java (hunk 2 above)
Comment (zoy0) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java]: done

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java (hunk 3 above)
Comment (zoy0) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ToBase64Function.java]: done

**Comment by aloyszhang** [comment]
please resolve the conflict.

**Comment by zoy0** [comment]
> please resolve the conflict.

done

**Review by vernedeng — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Review by yfsn666 — APPROVED** [review]

**Pull request merged**